corehq/apps/motech/openmrs/concepts/views.py

```diff
@@ -0,0 +1,50 @@
+from django.db.models import Q
+from django.http import JsonResponse
+from corehq.apps.motech.connected_accounts import get_openmrs_account
+from corehq.apps.motech.openmrs.concepts.models import OpenmrsConcept
+from corehq.apps.motech.openmrs.concepts.sync import \
+    openmrs_concept_json_with_answers_from_concept, sync_concepts_from_openmrs
+from corehq.apps.motech.permissions import require_motech_permissions
+
+
+@require_motech_permissions
+def all_openmrs_concepts(request, domain):
+    account = get_openmrs_account(domain)
+
+    concepts = OpenmrsConcept.objects.filter(account=account, openmrsconcept=None)
+    return JsonResponse({'concepts': [
+        openmrs_concept_json_with_answers_from_concept(concept).to_json()
+        for concept in concepts
+    ]})
+
+
+@require_motech_permissions
+def concept_search(request, domain):
+    search = request.GET.get('q') or ''
+    uuid = request.GET.get('uuid')
+    account = get_openmrs_account(domain)
+    if uuid:
+        try:
+            concept = OpenmrsConcept.objects.get(uuid=uuid, account=account)
+        except OpenmrsConcept.DoesNotExist:
+            return JsonResponse({'concepts': []})
+        else:
+            return JsonResponse({'concepts': [
+                openmrs_concept_json_with_answers_from_concept(concept).to_json()
+            ]})
+    elif len(search) > 2:
+        all_openmrs_concepts = OpenmrsConcept.objects.filter(Q(account=account) & ~Q(answers=None))
+        openmrs_concepts = all_openmrs_concepts.filter(names__icontains=search)
+        first_50 = openmrs_concepts[:50]
+        return JsonResponse({'concepts': [
```

Comment:
It doesn't; `JsonResponse` is a Django built-in that takes in a `dict` and returns a Django `HttpResponse` with the desired status code, a JSON body based on your dictionary, and content type set to the correct JSON mimetype (`application/json`). `.to_json` is a `JsonObject` method that converts from an instance of a `JsonObject` (subclass) to a JSON-serializable `dict` (not actually to JSON). I think it's a little confusing because at some point, possibly before Django's `JsonResponse` existed, we had defined a `json_response` helper that used our custom JSON handler that actually _does_ call the `.to_json()` method on the object-to-be-serialized, if it exists. This method is still used in some places, but I now prefer using the built-in `JsonResponse`. (Sorry for the characteristic over-explanation here!)